packages/frontend/src/providers/Dashboard/DashboardProvider.tsx

```diff
@@ -856,11 +856,30 @@ const DashboardProvider: React.FC<
             const setFunction = isTemporary
                 ? setDashboardTemporaryFilters
                 : setDashboardFilters;
-            setFunction((previousFilters) => ({
-                dimensions: [...previousFilters.dimensions, filter],
-                metrics: previousFilters.metrics,
-                tableCalculations: previousFilters.tableCalculations,
-            }));
+            if (isTemporary) {
```

Comment:
If I follow correctly, this would remove the saved filter which could cause issues.

I suggest updating logic to:

(if isTemporary is true)
1. detect if has a matching dashboard filter
2. if yes, override filter
3. if no, detect if has a matching dashboard temporary filter
4. if yes, override temporary filter
5. if no, add temporary filter

(if isTemporary is false)
1. detect if has a matching dashboard filter
2. if yes, override filter
3. if no, add filter